sorting/merge_sort.c

```diff
@@ -33,6 +33,11 @@ void swap(int *a, int *b)
 void merge(int *a, int l, int r, int n)
 {
     int *b = (int *)malloc(n * sizeof(int)); /* dynamic memory must be freed */
+    if (b == NULL) /* return to main if can't malloc memory */
+    {
+        printf("Can't Malloc! Please try again.");
+        exit(1);
```

Comment:
I would suggest you to use the macro instead of a magic number, although it probably does not matter given how widely `1` is recognized to be an abnormal exit code
```suggestion
        exit(EXIT_FAILURE);
```